packages/amplify-provider-awscloudformation/src/iterative-deployment/deployment-manager.ts

```diff
@@ -490,9 +490,18 @@ export class DeploymentManager {
     assert(stackParams.stackName, 'stackName should be passed to waitForDeployment');
 
     await cfnClient
-      .waitFor('stackUpdateComplete', {
-        StackName: stackParams.stackName,
-      })
+      .waitFor(
+        'stackUpdateComplete',
+        {
+          StackName: stackParams.stackName,
+        },
+        (err, data) => {
+          if (err) {
+            throw err;
+          }
+          return data;
+        },
+      )
```

Comment:
I believe this change is a no-op. Since the call is already awaited, just wrap it in try-catch and there throw `AmplifyFault`. See `waiter`s Error Handling here https://aws.amazon.com/blogs/developer/waiters-in-modular-aws-sdk-for-javascript/ and Amplify's converting this CFN error here https://github.com/aws-amplify/amplify-cli/blob/9ce1c4b30dd3a9e628eaa6c5e25658ade2179c43/packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js#L419-L423

Reply:
Good point. Updated!